test/algorithmiaTest.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const assert = window.chai ? window.chai.assert : require('assert');
+describe('Algorithmia', function() {
+    describe('client', function() {
+        before(async () => {
+            var api_key = window.location.search.slice(1).split('=')[1];
+            await  Algorithmia.query("demo/Hello", api_key , "HAL 9000", (error, result)=>{
+                document.getElementById('result').innerText = result;
+            });
+        });
+        it('should invoke algo endpoint',  async ()=> {
+           await  assert.equal(document.getElementById("result").innerText, "Hello HAL 9000");
+        });
+    });
+});
```

Comment:
Can simplify a bit if you'd like and assert on the returned value instead of interacting with the DOM.

```
'use strict';

const assert = window.chai ? window.chai.assert : require('assert');

describe('Algorithmia JS client', () => {
  it('should invoke algo endpoint', (done) => {
    const apiKey = window.location.search.slice(1).split('=')[1];

    Algorithmia.query(
      'demo/Hello',
      apiKey,
      'HAL 9000',
      (error, result) => {
        assert.equal(result, 'Hello HAL 9000');
        done();
      }
    );
  });
});
```

Reply:
ok, yes, sure

Reply:
yes, sure